### ISSUE TITLE
Can't focus Browse Data tables dropdown

### ISSUE THREAD

**Issue by revolter** [sqlitebrowser/sqlitebrowser#908]
### Details for the issue

Previously, after selecting a table and closing the dropdown, it remained focused, so you could hit the up/down arrow keys to change the table. Now, I can't focus it at all, even with multiple tab key presses.

#### I'm opening this issue because:

- [ ] DB4S is crashing
- [x] DB4S has a bug
- [ ] DB4S needs a feature
- [ ] DB4S has another problem

#### I'm using DB4S on:

- [ ] Windows: ( _version:_ ___ )
- [ ] Linux: ( _distro:_ ___ )
- [x] Mac OS: ( _version:_ Siera )
- [ ] Other: ___

#### I'm using DB4S version:

- [ ] 3.9.1
- [ ] 3.9.0
- [x] Other: Nightly 20.12.2016

#### I have also:

- [x] Tried out the nightly version: https://github.com/sqlitebrowser/sqlitebrowser#windows
- [ ] Searched for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
Interesting.  Does the nightly build from yesterday have the same problem?  If not, then it narrows the change down to something from yesterday, so easier to investigate. :smile:

**Comment by revolter** [comment]
No, it doesn't. Looks like tab index was removed from most of the controls.

**Comment by justinclift** [comment]
Interesting.  Nothing from yesterday's changes is really jumping out at me as a smoking gun.

Unless maybe [loadPragmas()](https://github.com/sqlitebrowser/sqlitebrowser/commit/ab5f6cffa702c26bc324e2b951818cbfb14dc462) is having some kind of unintended side effect.

Hmmm, going to assign to @MKleusberg, as it's likely his new code doing it. :wink:

**Comment by revolter** [comment]
Neither does the 1st of December Nightly work, so it's probably not something recent. A `git bisect` might help.

**Comment by justinclift** [comment]
Ahhh, I thought you meant "No, it doesn't." in response to "Does it have the same problem?".

But you're meaning it does have the same problem?

**Comment by revolter** [comment]
I'm dumb. I wanted to say that it has the problem too. I thought that maybe c221536d is the problem, but the 16th of August Nightly is working ok.

**Comment by justinclift** [comment]
Interesting.  You should be able to found out the exact day it started using this manual er... bisect-via-download approach. :smile:

**Comment by revolter** [comment]
I'll see if I can continue this at home 😆

**Comment by justinclift** [comment]
:smile:

**Comment by MKleusberg** [comment]
The tab stops and their order are stored in src/MainWindow.ui (somewhere at the bottom of the file). But I can't see any problem there and they haven't been touched lately.

What I noticed though is this: when I open a database and switch to the Browse Table tab and I can press Tab as much as I want and nothing ever happens. However, as soon as I click on one of the buttons, the combo box or the table view the Tab key starts to work again. Looks like this is how Qt is handling the Tab key on dialogs with a tab widget (any puns on this coincidence are very welcome btw). Is this the same for OS X? I've also pushed a commit which add the Alt+T shortcut to focus the combo box which should help a bit :)

**Comment by revolter** [comment]
> But I can't see any problem there and they haven't been touched lately.

That's what I knew about the tab stops and noticed too.

> However, as soon as I click on one of the buttons, the combo box or the table view the Tab key starts to work again

I remember trying this but it still didn't start working.

Will take a look on 4th of January so use the same pc + os version combination.

(there would be awesome if GitHub had a reminder feature)

**Comment by justinclift** [comment]
@revolter OSX → Applications → Reminders.app 😀

**Comment by chrisjlocke** [comment]
I know with VB.net, the same thing happens if you use a dataview grid. Tab
is used to shift focus to the next column in the grid and not the next
control. I've noticed the same when in the database maintenance grid ...
tab doesn't do much, so is it a grid thing and not a tab widget thing?




Thanks,
Chris

**Comment by revolter** [comment]
@MKleusberg I did the following:

- downloaded 04.01.2017 nightly
- opened database
- clicked `Browse Data`
- clicked refresh button
- hit <kbd>tab</kbd>

but nothing happened.

I somehow got it working but it behaved very weird:

I got the <kbd>tab</kbd> key to toggle the focus between controls, but when I kept it on the `New Record` and clicked the tables dropdown, the blue outline got removed, as it should, because the dropdown was focused. But when I closed the dropdown, the blue focus outline moved back to the `New Record` button instead of remaining on the dropdown.
